main/helpers.py

```diff
@@ -3,8 +3,8 @@
 
 def get_invalid_matches():
     matches = Match.objects.all()
```

Comment:
In general, this will not work for big databases. `.all()` executes and commits when it's called, which means that you cannot modify the query after this. Just imagine loading 3MM objects into memory with a single line of code. No good.

The recommendation here is to use the `.filter` function in Django ORM. You can build very precise queries with it and don't need to run logic on your database in memory.

Reply:
This is a good idea. I should probably switch to doing it this way.